guides/micronaut-oracle-autonomous-db/micronaut-oracle-autonomous-db.adoc

```diff
@@ -0,0 +1,235 @@
+
+Create a `DataPopulator` class to create some example database entries when the application starts:
+
+source:DataPopulator[]
+
+=== Controller class
+
+Create a `ThingController` class to view persisted data:
+
+source:controller/ThingController[]
+
+=== Configuration
+
+Replace the generated `application.yml` with this:
+
+resource:application.yml[]
+
+<1> Replace this temporary value with the database OCID unique identifier you saved when creating the database
+<2> Replace this temporary value with a password to encrypt the wallet keys (must be at least 8 characters and include at least 1 letter and either 1 numeric or special character)
+<3> Replace this temporary value with the `micronautdemo` schema user password you created
+<4> Change the profile name if you're not using the default, and optionally add a value for the path to the config file if necessary as described in the https://micronaut-projects.github.io/micronaut-oracle-cloud/latest/guide/#config-auth[Authentication section] of the Micronaut Oracle Cloud docs
+
+Replace the generated `application-test.yml` with this:
+
+testResource:application-test.yml[]
+
+Create a new Flyway migration SQL script in `src/main/resources/db/migration/V1__create-schema.sql` and add the following:
+
+resource:db/migration/V1__create-schema.sql[]
+
+And finally, you can optionally edit `src/main/resources/logback.xml` and add the following to monitor the SQL queries that Micronaut Data performs:
+
+[source,xml]
+----
+<logger name='io.micronaut.data.query' level='debug' />
+----
+
+=== Writing Tests
+
+Create a test to verify that database access works:
+
+test:ThingRepositoryTest[]
+
+include::{commondir}/common-testApp.adoc[]
+
+include::{commondir}/common-runapp.adoc[]
+
+You should see output similar to the following, indicating that the database connectivity and wallet configuration is all handled automatically, and the Flyway migration runs since the database is determined to be out of date. Also, if you added the Logback logger above, you'll see the results of the work done by `DataPopulator`:
+
+[source, bash]
+----
+INFO  com.oracle.bmc.Region - Loaded service 'DATABASE' endpoint mappings: {US_ASHBURN_1=https://database.us-ashburn-1.oraclecloud.com}
+INFO  c.oracle.bmc.database.DatabaseClient - Setting endpoint to https://database.us-ashburn-1.oraclecloud.com
+INFO  i.m.o.a.j.OracleWalletArchiveProvider - Using default serviceAlias: MicronautDemo_high
+INFO  i.m.flyway.AbstractFlywayMigration - Running migrations for database with qualifier [default]
+INFO  o.f.c.i.database.base.DatabaseType - Database: jdbc:oracle:thin:@(description=(retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1522)(host=adb.us-ashburn-1.oraclecloud.com))(connect_data=(service_name=rxfmolsmtfaakhf_micronautdemo_high.adb.oraclecloud.com))(security=(ssl_server_cert_dn="CN=adwc.uscom-east-1.oraclecloud.com,OU=Oracle BMCS US,O=Oracle Corporation,L=Redwood City,ST=California,C=US"))) (Oracle 19.0)
```

Comment:
Not sure if this contains any sensitive information or not.

Reply:
nothing sensitive, and I'll be terminating the instance before the guide is published anyway

Reply:
ok